README.md

```diff
@@ -37,51 +37,102 @@ brane import wscs-2021/vis-utils
+```
+plot(input_path, 
+    <optional> x_axis, 
+    <optional> y_axis, 
+    <optional> hue_axis, 
+    <optional> options, 
+    <optional> output_path)
+```
+
+### Examples
+
+> Empty strings `""` indicate `None`.
+
+Plot all:
 
 ```sh
 > import vis-utils;
-> plot("path/to/input.csv", (optional)"<column_name>", (optional)"<column_name>", (optional)"<column_name>", (optional) "options", (optional), (optional) "path/to/output.csv")
+> plot("path/to/input.csv", 
+      "", 
+      "", 
+      "", 
+      "", 
+      ""
+  )
+```
+
+Plot with options:
+
+```sh
+> import vis-utils;
+> plot("path/to/input.csv", 
+      "", 
+      "", 
+      "", 
+      [
+        [['species', 'flipper_length_mm', 'sex'], 'violinplot'],
+        [['species', 'bill_length_mm'], 'barplot']
+      ],
+      ""
+  )
+```
+
+Plot with axes given:
+
+```sh
+> import vis-utils;
+> plot("path/to/input.csv", 
+      "species", 
+      "island", 
+      "", 
+      "",
+      ""
+  )
 ```
-**Input**:
-- REQUIRED `input_path`: string, path to .csv/.json/.parquet data file
-    - Example: "tests/dummy_data/penguins.csv"
 
+### Arguments
 
+- REQUIRED `input_path`: string, path to .csv/.json/.parquet data file
+    - Example: "tests/dummy_data/penguins.csv"
 - OPTIONAL `x_axis`: string, column name of desired x_axis
     - Example: "species"
+    - Default: None
 - OPTIONAL `y_axis`: string, column name of desired y_axis
     - Example: "flipper_length_mm"
+    - Default: None
 - OPTIONAL `hue_axis`: string, column name of desired hue_axis
     - Example: "sex"
-  
-
+    - Default: None
 - OPTIONAL `options`: string, list of sublists, containing [subsublist of column names] and a plot name 
     - Example: `"[[['species', 'sex'], 'barplot'],[['species', 'flipper_length_mm'],'violinplot']]"`
     - Possible plot names: `'relplot', 'scatterplot', 'lineplot', 'displot', 'histplot', 'kdeplot',
       'ecdfplot', 'rugplot', 'catplot', 'stripplot', 'swarmplot', 'boxplot',
       'violinplot', 'pointplot', 'barplot'`
-
-
+    - Default: None
 - OPTIONAL `output_path`: string, path to directory where figures will be written
     - Example: "tests"
+    - Default: None
+
+### Notes
 
-**Notes**:
+- Omitting an optional parameter can be done by passing an empty string `""`, e.g. `plot("path/to/input.csv", "species", "island", "", "" ,"")`
 - If no optional params are given, the function automatically plots all possible columns
 - If options is given, axes will be disregarded
-- not all axes need to be specified, all three can be combined in any way (e.g. x_axis + hue_axis)
-- If output_path is not specified, it will output to the same directory as the input file
+- Not all axes need to be specified, all three can be combined in any way (e.g. pass only x_axis and hue_axis)
+- If `output_path` is not specified, it will output to the same directory as the `input_path`
+
+### Output
```

Comment:
Maybe change it to Returns? Either way it's fine though

Reply:
Yeah we could. Even though output might be a bit more accurate, since we are not strictly returning any plotted `.png` files, but we are 'outputting' them I suppose?

Reply:
True, I'll merge.